### PR TITLE
Add note to README regarding building uber-jars

### PR DESCRIPTION
**Pull request by Marcono1234**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -117,6 +117,18 @@ Capturing a context snapshot and reactivating it in another thread is no differe
 For insight, the library tracks the overall time used creating and reactivating
 context snapshots along with time spent in each individual `ContextManager`.
 
+## Building uber-jars / jars with dependencies
+
+When using a build tool or plugin to create an 'uber-jar', i.e. a jar file with all
+the classes of its dependencies included, you have to make sure that the service
+provider-configuration files under `META-INF/services` are preserved, respectively
+merged. Otherwise Java's `ServiceLoader` will not be able to find the context
```
Comment (sjoerdtalsma) [inline readme.md:125]: _are preserved, respectively merged_ --> _are either preserved or merged_

File: readme.md [hunk 2]
```diff
@@ -117,6 +117,18 @@ Capturing a context snapshot and reactivating it in another thread is no differe
 For insight, the library tracks the overall time used creating and reactivating
 context snapshots along with time spent in each individual `ContextManager`.
 
+## Building uber-jars / jars with dependencies
```
Comment (sjoerdtalsma) [inline readme.md:120]: Can you move this section just above `Performance metrics`, so that isn't separated from logging performance and metrics reporting?

**Review by sjoerdtalsma — CHANGES_REQUESTED** [review]
Thanks for this addition!

Some small requests, but otherwise good to go 👍

File: readme.md [hunk 3]
```diff
@@ -117,6 +117,18 @@ Capturing a context snapshot and reactivating it in another thread is no differe
...
+
+When using a build tool or plugin to create an 'uber-jar', i.e. a jar file with all
+the classes of its dependencies included, you have to make sure that the service
+provider-configuration files under `META-INF/services` are preserved, respectively
```
[the remark this answers is not in the document]
Comment (Marcono1234) [reply readme.md:124]: "provider-configuration file" is the term used by the [`ServiceLoader`](https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/util/ServiceLoader.html), section "Deploying service providers on the class path".

File: readme.md (hunk 2 above)
Comment (Marcono1234) [reply readme.md]: Ah, sorry yes that paragraph is misplaced.

**Review by sjoerdtalsma — APPROVED** [review]
Thanks for the quick resolution, much appreciated!

**Pull request merged**